### PR TITLE
Armada-458 Improve logging in Jobbergate API

### PR DESCRIPTION
**Pull request by fschuch**
#### What
Improve logging in Jobbergate-API.

#### Why
We are not logging much in the Jobbergate-API. This makes it very difficult to triage problems that arise. We should add more debug logging to the API so that we can better tell where issues are happening.

`Task`: https://app.clickup.com/t/18022949/ARMADA-458

---

#### Peer Review
Please follow the upstream omnivector documentation concerning
[peer-review guidelines](https://github.com/omnivector-solutions/Documentation/blob/main/Contributing/pr_review_standards.md#peer-review).

### CODE REVIEW

**Comment by dusktreader** [comment]
Task linked: [ARMADA-458 Improve logging in jobbergate-api](https://app.clickup.com/t/18022949/ARMADA-458)

File: jobbergate-api/jobbergate_api/apps/applications/routers.py [hunk 1]
```diff
@@ -98,16 +119,20 @@ async def applications_delete_upload(
...
-            detail=f"Application {application_id=} not found.",
-        )
+        message = f"Application {application_id=} not found."
+        logger.warning(message)
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=message)
+
     application = ApplicationResponse.parse_obj(raw_application)
 
     if not application.application_uploaded:
+        logger.warning(f"Trying to delete a applications that was not uploaded ({application_id=})")
```
Comment (heitorPB) [inline jobbergate-api/jobbergate_api/apps/applications/routers.py:135]: The return code is 2XX, part of the "OK" class. It seems a bit too scary to send a warning as a log. I think `debug` is an appropriate level here.

Is this case (when entering the `if` block) an issue when using jobbergate?

File: jobbergate-api/jobbergate_api/apps/applications/routers.py [hunk 2]
```diff
@@ -134,22 +161,31 @@ async def application_delete(
...
     await database.execute(delete_query)
     try:
         del s3man_applications[application_id]
     except KeyError:
-        # We should ignore KeyErrors from the S3 manager, because the data may have already been removed
-        # outside of the API
-        pass
+        # We should ignore KeyErrors from the S3 manager, because the data may
+        # have already been removed outside of the API
+        logger.warning(f"Tried to delete {application_id=}, but it was not found on S3.")
```
Comment (heitorPB) [inline jobbergate-api/jobbergate_api/apps/applications/routers.py:188]: Shouldn't this be a debug as well?

File: jobbergate-api/jobbergate_api/apps/applications/routers.py (hunk 1 above)
Comment (fschuch) [reply jobbergate-api/jobbergate_api/apps/applications/routers.py:135]: I agree, I'll change it to `debug`.

In this case, the `id` is provided by the users. If they want to delete something that was not uploaded or was already deleted, we can just stop the function.

File: jobbergate-api/jobbergate_api/apps/applications/routers.py (hunk 2 above)
Comment (fschuch) [reply jobbergate-api/jobbergate_api/apps/applications/routers.py:188]: Yes

File: jobbergate-api/jobbergate_api/apps/applications/routers.py [hunk 3]
```diff
@@ -38,19 +45,27 @@ async def applications_create(
...
 
+    logger.debug("Inserting application")
+
     try:
         insert_query = applications_table.insert().returning(applications_table)
+        logger.debug(f"insert_query = {render_sql(insert_query)}")
         application_data = await database.fetch_one(query=insert_query, values=create_dict)
 
     except INTEGRITY_CHECK_EXCEPTIONS as e:
+        logger.warning(f"INTEGRITY_CHECK_EXCEPTIONS: {str(e)}")
```
Comment (matheushent) [inline jobbergate-api/jobbergate_api/apps/applications/routers.py:64]: shouldn't it be logging.error?

File: jobbergate-api/jobbergate_api/apps/applications/routers.py (hunk 1 above)
Comment (matheushent) [inline jobbergate-api/jobbergate_api/apps/applications/routers.py:135]:
```suggestion
        logger.warning(f"Trying to delete an application that was not uploaded ({application_id=})")
```

File: jobbergate-api/jobbergate_api/apps/job_scripts/routers.py [hunk 4]
```diff
@@ -134,15 +150,18 @@ async def job_script_create(
 
     Make a post request to this endpoint with the required values to create a new job script.
     """
-    logger.debug(f"Creating job_script with: {job_script}")
+    logger.debug(f"Creating {job_script=}")
+
     select_query = applications_table.select().where(applications_table.c.id == job_script.application_id)
+    logger.debug(f"select_query = {render_sql(select_query)}")
```
[the remark this answers is not in the document]
Comment (fschuch) [reply jobbergate-api/jobbergate_api/apps/job_scripts/routers.py:156]: Logging the queries is a point open to discussion. We can use `logger.trace` to activate them in very specific situations or remove them for now, what do you think?

File: jobbergate-api/jobbergate_api/apps/applications/routers.py (hunk 2 above)
Comment (fschuch) [reply jobbergate-api/jobbergate_api/apps/applications/routers.py]: solved in [0c61e5d](https://github.com/omnivector-solutions/jobbergate/pull/121/commits/0c61e5d42e5c10b8193ec444d45138c631f1b87a)

File: jobbergate-api/jobbergate_api/apps/applications/routers.py (hunk 1 above)
Comment (fschuch) [reply jobbergate-api/jobbergate_api/apps/applications/routers.py]: solved in [0c61e5d](https://github.com/omnivector-solutions/jobbergate/pull/121/commits/0c61e5d42e5c10b8193ec444d45138c631f1b87a)

Comment (fschuch) [reply jobbergate-api/jobbergate_api/apps/applications/routers.py]: solved in [0b4f677](https://github.com/omnivector-solutions/jobbergate/pull/121/commits/0b4f677de19511164ead7888b0e8b72006ea43e1)

File: jobbergate-api/jobbergate_api/apps/applications/routers.py (hunk 3 above)
Comment (fschuch) [reply jobbergate-api/jobbergate_api/apps/applications/routers.py]: solved in [0b4f677](https://github.com/omnivector-solutions/jobbergate/pull/121/commits/0b4f677de19511164ead7888b0e8b72006ea43e1)

File: jobbergate-api/jobbergate_api/apps/job_scripts/routers.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (matheushent) [reply jobbergate-api/jobbergate_api/apps/job_scripts/routers.py:156]: well, let's go with logger.trace IMO

[the remark this answers is not in the document]
Comment (fschuch) [reply jobbergate-api/jobbergate_api/apps/job_scripts/routers.py]: solved in [2a746b7](https://github.com/omnivector-solutions/jobbergate/pull/121/commits/2a746b78551d6720dd18853db5b7cfbd9cd044a3)

**Review by matheushent — APPROVED** [review]

**Pull request merged**